rust/scannerctl/src/interpret/mod.rs

```diff
@@ -171,6 +174,23 @@ where
     Ok(result)
 }
 
+fn create_loader_from_feed_json(
+    store: &DefaultDispatcher,
+    path: PathBuf,
+) -> Result<FSPluginLoader<PathBuf>, CliError> {
```

Comment:
To return a FSPluginLoader doesn't make sense here, as the given path is a path to a json file and hence cannot be used as a FSPLuginLoader. An alternative would be either:
- return FSPluginLoader pointing to path.parent()
- don't return a FSPluginLoader and change interprete::mod::run method to call `set_vts` when it's a json and only otherwise return a FSPluginLoader.

Reply:
I think I would rewrite `create_loader_from_feed_json` like:
```
fn create_loader_from_feed_json(
    store: &DefaultDispatcher,
    path: PathBuf,
) -> Result<FSPluginLoader<PathBuf>, CliError> {
    let file = File::open(&path)?;
    let mut reader = BufReader::new(file);
    let vts: Vec<storage::item::Nvt> = serde_json::from_reader(&mut reader)?;
    let all_vts = vts.into_iter().map(|v| (v.filename.clone(), v)).collect();

    store.set_vts(all_vts).map_err(|e| CliError {
        filename: path.to_owned().to_string_lossy().to_string(),
        kind: e.into(),
    })?;
    match path.parent() {
        None => Err(CliError {
            filename: path.to_str().map(|x| x.to_string()).unwrap_or_default(),
            kind: CliErrorKind::Corrupt("Does not have a parent".to_string()),
        }),
        Some(p) => Ok(FSPluginLoader::new(p.to_path_buf())),
    }
}
```

That way the FSPluginLoader shows to the parent of the json feed meta information which in most scenarios should be correct.However I think moving this functionality to the DefaultDispatcher and replace `set_vts` to `load_from_feed_json` using that logic there would be better as we could reconstruct the run method to use the feed path to construct the FSPluginLoader and it has a feed.json or so then it can load it without having it to parse each and every script but can fallback to it.

That way we could reuse this functionality when there is a proper feed that has the json in it later on.